tests/commands/release-asset/download.spec.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license
+ * Copyright 2025 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import * as fs from 'fs';
+import * as nock from 'nock';
+import * as path from 'path';
+import * as os from 'os';
+import { BalenaAPIMock } from '../../nock/balena-api-mock';
+import { cleanOutput, runCommand } from '../../helpers';
+
+// "itSS" means "it() Skip Standalone"
+const itSS = process.env.BALENA_CLI_TEST_TYPE === 'standalone' ? it.skip : it;
+
+describe('balena release-asset download', function () {
+	let api: BalenaAPIMock;
+	let tempDir: string;
+
+	beforeEach(async () => {
+		api = new BalenaAPIMock();
+		api.expectGetWhoAmI({ optional: true, persist: true });
+
+		tempDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'balena-test-'));
+	});
+
+	afterEach(async () => {
+		await fs.promises.rm(tempDir, { recursive: true, force: true });
+		api.done();
+	});
+
+	itSS('should download a release asset successfully', async () => {
+		const testContent = 'test file content for download';
+
+		api.expectGetRelease();
+		api.expectGetReleaseAsset({ found: true, assetId: 123 });
```

Comment:
nit since `123` is the default
```suggestion
		api.expectGetReleaseAsset({ found: true });
```